tools/make_patch.c

```diff
@@ -0,0 +1,944 @@
+			buffer_index = SIZE_MAX;
+			offset = parse_abs_offset(bank, address, buffer[0]);
+			if (offset == -1) break;
+
+			if (create_symbol(&symbols, offset, buffer)) goto error;
+
+			offset = -1;
+			break;
+
+		case ' ':
+			// If we've just encountered a space, ignore this one
+			if (offset != -1) {
+				if (buffer_index) {
+					buffer[buffer_index++] = c;
+				}
+				break;
+			}
+
+			// If we encounter a space, we've reached the end of the address.
+			buffer[buffer_index] = '\0';
+			bank = parse_address_bank(buffer);
+			address = parse_address_address(buffer);
+			buffer_index = offset ? 0 : SIZE_MAX;
+
+			break;
+
+		default:
+			buffer[buffer_index++] = c;
+		}
+
+		// We were requested to skip to the next line
+		if (buffer_index == SIZE_MAX) {
+			buffer_index = 0;
+			offset = -1;
+
+			while (c != '\n') {
+				if ((c = getc(file)) == EOF) break;
+			}
+		}
+	}
+
+	if (!symbols) {
+		fprintf(stderr, "Error: No symbols found.");
+	}
+
+	free_buffer(buffer);
+	return symbols;
+
+error:
+	if (buffer) free_buffer(buffer);
+	return NULL;
+}
+
+int parse_rgbds_int(char *string)
+{
+	char *start = string;
+	char *end;
+	int base = 10;
+	int value;
+
+	// Figure out the base
+	if (*start == '$') {
+		base = 16;
+		start++;
+	}
+
+	errno = 0;
+	value = strtol(start, &end, base);
+
+	if (errno || end != string + strlen(string)) {
+		// Parsing complex things is out of this program's scope.
+		// Warning the user would be a good thing, but warning
+		// for things we just can't parse would be annoying...
+		return -1;
+	}
+
+	return value;
+}
+
+struct symbol *parse_asm(FILE *file)
+{
+	struct symbol *symbols = NULL;
+	char *label = NULL;
+	char *buffer;
+	size_t buffer_index = 0;
+	int const_value = -1;
+	int c;
+
+	enum {
+		PARSING_LABEL,
+		PARSING_INSTRUCTION,
+		PARSING_EQU,
+		PARSING_CONST_DEF,
+		PARSING_CONST
+	} parsing = PARSING_LABEL;
+
+	buffer = create_buffer();
+	if (!buffer) goto error;
+
+	// Get all the constants defined in a file
+	while ((c = getc(file)) != EOF) {
+		buffer = expand_buffer(buffer, buffer_index);
+		if (!buffer) goto error;
+
+		switch(c) {
+		case ' ':
+		case '\t':
+			// Eat these characters if we haven't picked up anything else yet
+			if (!buffer_index) {
+				// If the line starts with one of these characters,
+				// we're parsing an instruction.
+				if (parsing == PARSING_LABEL) parsing = PARSING_INSTRUCTION;
+				break;
+			}
+
+			if (parsing == PARSING_LABEL) {
+				// A label starts with its name
+				buffer[buffer_index] = '\0';
+				buffer_index = 0;
+				label = malloc(strlen(buffer) + 1);
+				if (!label) {
+					fprintf(stderr, "Error: Cannot allocate memory\n");
+					goto error;
+				}
+				strcpy(label, buffer);
+				parsing = PARSING_INSTRUCTION;
+				break;
+			} else if (parsing == PARSING_INSTRUCTION) {
+				// Figure out what instruction we've encountered
+				buffer[buffer_index] = '\0';
+				buffer_index = SIZE_MAX;
+
+				if (label && strcmp(buffer, "EQU") == 0) {
```

Comment:
Here and below, `!strcmp(a, b)` is shorter than `strcmp(a, b) == 0`, and is what other tools use.
```suggestion
				if (label && !strcmp(buffer, "EQU")) {
```

Reply:
Resolved in https://github.com/pret/pokecrystal/pull/882/commits/1655f032f570fd25c18c3836781be2d830c3032d